src/common/utils/paths.ts

```diff
@@ -33,3 +33,35 @@ function resolveTilde(filePath: string) {
 export function resolvePath(filePath: string): string {
   return path.resolve(resolveTilde(filePath));
 }
+
+/**
+ * Checks if `testPath` represents a potential filesystem entry that would be
+ * logically "within" the `parentPath` directory.
+ *
+ * This function will return `true` in the above case, and `false` otherwise.
+ * It will return `false` if the two paths are the same (after resolving them).
+ *
+ * The function makes no FS calls and is platform dependant. Meaning that the
+ * results are only guaranteed to be correct for the platform you are running
+ * on.
+ * @param parentPath The known path of a directory
+ * @param testPath The path that is to be tested
+ */
+export function isLogicalChildPath(parentPath: string, testPath: string): boolean {
+  parentPath = path.resolve(parentPath);
+  testPath = path.resolve(testPath);
+
+  if (parentPath === testPath) {
+    return false;
+  }
+
+  while (testPath.length >= parentPath.length) {
```

Comment:
Instead of while loop why not just `testPath.startsWith(parentPath)`?

Reply:
Because of the following case:

```ts
const testPath = "/foo/bar";
const parentPath = "/fo";
```

Reply:
`testPath.startsWith(parentPath + path.sep)`?

Reply:
Nope because on windows both \ and / are valid separators. I could check if I am on windows first though